canarytokens/channel_input_mysql.py

```diff
@@ -38,25 +38,30 @@ def dataReceived(self, data: bytes) -> None:
         try:
             canarydrop, hit = self.handleQuery(src_host, self.buf)
             self.factory.dispatch(canarydrop=canarydrop, token_hit=hit)
-            print(f"alert dispatched for token {canarydrop.canarytoken.value()}: {hit}")
-        except (NoCanarytokenFound, Exception) as e:
+        except (ValueError, NoCanarytokenFound, NoCanarytokenPresent) as e:
+            log.info(f"{e}")
+        except Exception as e:
             log.error(f"Error in MySQL channel: {e} | Data received: {data}")
         self.transport.loseConnection()
 
     @staticmethod
     def handleQuery(src_host: str, buf: bytes) -> Tuple[Canarydrop, MySQLTokenHit]:
-        capability_flags, max_packet_sz, char_set, username = struct.Struct(
-            "!4s4sc27x25s"
-        ).unpack_from(buf)
+        try:
+            capability_flags, max_packet_sz, char_set, username = struct.Struct(
+                "!4s4sc27x25s"
+            ).unpack_from(buf)
+        except struct.error:
+            raise ValueError("Failed to decode MySQL query.")
 
         log.info(f"MySQL Query from {src_host}: {username}")
-
-        additional_info = CanaryMySQLProtocol.additionalInfo(buf)
-        print(f"mysql token additional_info: {additional_info}")
         token = Canarytoken(value=username)
         canarydrop = get_canarydrop(canarytoken=token)
-        if not canarydrop:
-            raise ValueError("Canarydrop not found")
+
+        try:
+            additional_info = CanaryMySQLProtocol.additionalInfo(buf)
+        except Exception as e:
+            log.info(f"Error getting additional info: {e} from buf: {buf}")
```

Comment:
We should be using `log.exception`. It is a really nice way to log the exception with the stack trace and everything included.
It would look like:
```
log.exception(f"Error getting additional info from buf: {buf}")
```